packages/dialog/src/mwc-dialog-base.ts

```diff
@@ -118,7 +120,60 @@ export class DialogBase extends BaseElement {
   }
 
   protected getInitialFocusEl(): HTMLElement|null {
-    return this.querySelector(`[${this.initialFocusAttribute}]`);
+    const initFocusSelector = `[${this.initialFocusAttribute}]`;
+
+    // only search light DOM. This typically handles all the cases
+    const lightDomQs = this.querySelector(initFocusSelector);
+
+    if (lightDomQs) {
+      return lightDomQs as HTMLElement;
+    }
+
+    // if not in light dom, search each flattened distributed node.
+    const primary = this.primarySlot as HTMLSlotElement;
+    const secondary = this.secondarySlot as HTMLSlotElement;
```

Comment:
This property access calls the query selector, so I think you should move this, and the content one, down to where they are used too, so that we can short-circuit them.